### PR TITLE
Mark `slice::copy_from_slice` unstably const

### PR DESCRIPTION
**Pull request by okaneco**
Tracking issue #131415

I used `const_eval_select` for runtime and const panic functions because const formatting isn't available yet.

### CODE REVIEW

**Comment by RalfJung** [comment]
r? libs-api

File: library/core/src/slice/mod.rs [hunk 1]
```diff
@@ -3681,11 +3682,21 @@ impl<T> [T] {
...
+        const fn len_mismatch_fail(dst_len: usize, src_len: usize) -> ! {
+            const fn panic_at_const(_dst_len: usize, _src_len: usize) -> ! {
+                // Note that we cannot format in constant expressions.
+                panic!(
+                    "copy_from_slice: source slice length does not match destination slice length"
+                );
+            }
+            fn panic_at_rt(dst_len: usize, src_len: usize) -> ! {
+                panic!(
+                    "source slice length ({}) does not match destination slice length ({})",
```
Comment (bjoernager) [inline library/core/src/slice/mod.rs:3694]: Add the `copy_from_slice: ` part like in `panic_at_const` ?

Comment (bjoernager) [reply library/core/src/slice/mod.rs:3694]: Also perhaps inline the format arguments:

```rust
panic!(
    "copy_from_slice: source slice length ({src_len}) does not match destination slice length ({dst_len})",
);
```

**Comment by alex-semenyuk** [comment]
@okaneco 
Could you please take a look comments?

**Comment by RalfJung** [comment]
This is unstable (i.e., we can take it back if needed) and should be uncontroversial, so let me just

@bors r+

**Pull request merged**